src/modules/Popup/Popup.js

```diff
@@ -0,0 +1,270 @@
+      return 'top center'
+    }
+  }
+
+  getPopupStyle() {
+    if (!this.coords || !this.popupCoords) return
+    let position = this.props.position
+    let style = this.computePopupStyle(position)
+
+    // Lets detect if the popup is out of the viewport and adjust
+    // the position accordingly
+    const correctedPosition = this.correctPositionForViewPort(style)
+    if (correctedPosition) {
+      style = this.computePopupStyle(correctedPosition)
+      // Set the position so the carret will appear at the right place
+      position = correctedPosition
+    }
+
+    // Append 'px' to every numerical values in the style
+    style = _.mapValues(style, value => _.isNumber(value) ? value + 'px' : value)
+
+    this.setState({ style, position })
+  }
+
+  getPortalProps() {
+    const portalProps = { onOpen: this.onOpen }
+    const { on, hoverable } = this.props
+
+    switch (on) {
+      case 'click':
+        portalProps.openOnTriggerClick = true
+        portalProps.closeOnTriggerClick = true
+        portalProps.closeOnDocumentClick = true
+        break
+
+      case 'focus':
+        portalProps.openOnTriggerFocus = true
+        portalProps.closeOnTriggerBlur = true
+        break
+
+      default:  // default to hover
+        portalProps.openOnTriggerMouseOver = true
+        portalProps.closeOnTriggerMouseLeave = true
+        // Taken from SUI: https://git.io/vPmCm
+        portalProps.mouseLeaveDelay = 70
+        portalProps.mouseOverDelay = 50
+        break
+    }
+
+    if (hoverable) {
+      portalProps.closeOnPortalMouseLeave = true
+      portalProps.mouseLeaveDelay = 300
+    }
+
+    return portalProps
+  }
+
+  hideOnScroll = (event) => {
+    this.setState({ closed: true })
+    window.removeEventListener('scroll', this.hideOnScroll)
+    setTimeout(() => this.setState({ closed: false }), 50)
+  }
+
+  onOpen = (event) => {
+    this.coords = event.currentTarget.getBoundingClientRect()
+    if (this.props.hideOnScroll) {
+      window.addEventListener('scroll', this.hideOnScroll)
+    }
+  }
+
+  popupMounted = (ref) => {
+    this.popupCoords = ref ? ref.getBoundingClientRect() : null
+    this.getPopupStyle()
+  }
+
+  render() {
+    const {
+      basic,
+      children,
+      className,
+      content,
+      flowing,
+      fluid,
+      header,
+      inverted,
+      size,
+      trigger,
+      width,
+    } = this.props
+
+    const { position, closed } = this.state
+
+    const style = _.assign({}, this.state.style, this.props.style)
+
+    const classes = cx(
+      className,
+      position,
+      size,
+      width,
+      'ui popup transition visible',
+      {
+        basic,
+        flowing,
+        fluid,
+        inverted,
+      }
+    )
+
+    if (closed) return trigger
+
+    return (
+      <Portal trigger={trigger} {...this.getPortalProps()}>
+        <div ref={this.popupMounted} style={style} className={classes} >
```

Comment:
I think there might be `ElementType` from `getElementType` instead of `div`.

```jsx
const ElementType = getElementType(Popup, props)

<ElementType>...</ElementType>
```